js/twister_formatpost.js

```diff
@@ -379,196 +404,228 @@ function htmlFormatMsg(msg, mentions) {
                 if (p[i].a > -1) {
                     if (p[i].t === -1 || (p[i].t === 0 && p[i].a > i)) {
                         if (p[i].k > 1)
-                            html.push('<' + tag + '>' + Array(p[i].k).join(chr));
+                            msg.htmlEntities.push('<' + tag + '>' + Array(p[i].k).join(chr));
                         else
-                            html.push('<' + tag + '>');
+                            msg.htmlEntities.push('<' + tag + '>');
                     } else if (p[i].t === 1 || (p[i].t === 0 && p[i].a < i)) {
                         if (p[i].k > 1)
-                            html.push(Array(p[i].k).join(chr) + '</' + tag + '>');
+                            msg.htmlEntities.push(Array(p[i].k).join(chr) + '</' + tag + '>');
                         else
-                            html.push('</' + tag + '>');
+                            msg.htmlEntities.push('</' + tag + '>');
                     }
-                        strEncoded = '>' + (html.length - 1).toString() + '<';
-                        str = str.slice(0, p[i].i) + strEncoded + str.slice(p[i].i + p[i].k);
-                        l = strEncoded.length - p[i].k;
+                        htmlEntityEncoded = '>' + (msg.htmlEntities.length - 1).toString() + '<';
+                        msg.str = msg.str.slice(0, p[i].i) + htmlEntityEncoded + msg.str.slice(p[i].i + p[i].k);
+                        l = htmlEntityEncoded.length - p[i].k;
                         for (j = i + 1; j < p.length; j++)
                             p[j].i += l;
                 }
             }
         }
 
-        return str;
+        return msg;
     }
 
-    function htmlSplitCounter(str) {
-        html.push('<span class="splited-post-counter">' + str + '</span>');
+    function newHtmlEntityLink(template, urlTarget, urlName) {
+        template.href = urlTarget;
+        template.innerHTML = urlName;  // .innerHTML instead .text to allow markup inside [] of [url name](target)
```

Comment:
damn good paranoid question! answer is: it depends, sanitization is various but looks like enough.

newHtmlEntityPlaintext is would be used to set mentions, hashtags, plain text links and mails URL text only. but as I see there is no need to additional escaping in these cases. and then newHtmlEntityLinkFormatted which does not escape somthing is needed only for formatted links — exactly where we are feared to meet some unescaped stuff. so this separation into two functions is pointless.

here is some problem with handling links definition inside links definition, see [comment in the code below](https://github.com/miguelfreitas/twister-html/pull/272/files#diff-b2c9a32ea5689959c8b839a3cb36827eR481). e.g. see @thedod's k=2372:

`@tasty [![this is fun](http://f.666kb.com/i/d0d8vpbfw3w0rcq6r.jpg)](http://www.phrases.org.uk/meanings/push-the-envelope.html) 😈`

which is turning into:

`<a href="#profile?user=tasty" class="open-profile-modal">@tasty</a> [!<a href="http://f.666kb.com/i/d0d8vpbfw3w0rcq6r.jpg" rel="nofollow" target="_blank">this is fun</a>](<a href="http://www.phrases.org.uk/meanings/push-the-envelope.html" rel="nofollow" target="_blank">http://www.phrases.org.uk/meanings/push-the-envelope.html</a>) 😈`

so browsers render it like this:

`@tasty [!this is fun](http://www.phrases.org.uk/meanings/push-the-envelope.html) 😈`

where `this is fun` and http://www.phrases.org.uk/meanings/push-the-envelope.html  is links.

I suggest to pull this as is and then fix continiously like everything else we got.